lib/internal/promisify.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+module.exports = promisify;
+
+const promises = require('internal/promises');
+const BuiltinPromise = promises.Promise;
+
+function promisify (fn, single) {
+  Object.defineProperty(inner, 'name', {
+    writable: false,
+    enumerable: false,
+    configurable: true,
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner () {
```

Comment:
Btw, can just omit everything apart from the `value` in descriptor (it will be inherited anyway).